src/include/duckdb/catalog/catalog_entry/index_catalog_entry.hpp

```diff
@@ -24,21 +23,40 @@ class IndexCatalogEntry : public StandardEntry {
 	static constexpr const char *Name = "index";
 
 public:
-	//! Create an IndexCatalogEntry and initialize storage for it
+	//! Create an IndexCatalogEntry
 	IndexCatalogEntry(Catalog &catalog, SchemaCatalogEntry &schema, CreateIndexInfo &info);
 
-	optional_ptr<Index> index;
+	//! The SQL of the CREATE INDEX statement
 	string sql;
+	//! Additional index options
+	case_insensitive_map_t<Value> options;
+
+	//! We need the initial size of the index after the CREATE INDEX statement,
+	//! as it is necessary to determine the auto checkpoint threshold
+	idx_t initial_index_size;
```

Comment:
This feels like it should be an implementation property - it really depends on how the catalog is implemented.  It might not be written to the duckdb storage (and thus checkpointing is not relevant). Should it be a DuckIndexEntry property?